### PR TITLE
Selinux in permissions

### PR DESCRIPTION
**Pull request by jelly**
### Depends

* [x] https://github.com/cockpit-project/cockpit-files/pull/775

## files

image

## Directories

image

### CODE REVIEW

File: src/dialogs/permissions.jsx [hunk 1]
```diff
@@ -90,6 +95,15 @@ const EditPermissionsModal = ({ dialogResult, selected, path }) => {
...
             console.error("Cannot obtain users from getent group", exc);
         }
+
+        try {
+            const full_path = selected?.isCwd ? path : path + selected.name;
+            const selinux_context = await python.spawn(read_selinux_context, full_path);
+            setSELinuxContext(selinux_context);
+        } catch (exc) {
+            if (exc.exit_status !== 2)
+                console.error("Cannot obtain SELinux context", exc);
```
Comment (cockpituous) [inline src/dialogs/permissions.jsx:105]: These 3 added lines are not executed by any test.

File: src/dialogs/permissions.jsx [hunk 2]
```diff
@@ -31,12 +32,15 @@ import { InlineNotification } from 'cockpit-components-inline-notification';
...
 import { useInit } from 'hooks';
 import { etc_group_syntax, etc_passwd_syntax } from 'pam_user_parser';
+import * as python from "python.js";
 import { superuser } from 'superuser';
 import { fmt_to_fragments } from 'utils.tsx';
 
 import { useFilesContext } from '../app.tsx';
 import { inode_types } from '../common.ts';
 
+import read_selinux_context from './read-selinux.py';
```
Comment (tomasmatus) [inline src/dialogs/permissions.jsx:42]: ok 🪄

File: src/dialogs/read-selinux.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/python3
+
+import os
+import os.path
```
Comment (tomasmatus) [inline src/dialogs/read-selinux.py:4]: do we need both `os` and `os.path`?

**Review by tomasmatus — CHANGES_REQUESTED** [review]
Just some minor nitpicks and me being surprised by the existence of `python.js` 👍🏻

File: src/dialogs/permissions.jsx (hunk 1 above)
Comment (cockpituous) [inline src/dialogs/permissions.jsx:105]: These 3 added lines are not executed by any test.

Comment (cockpituous) [inline src/dialogs/permissions.jsx:105]: These 3 added lines are not executed by any test.

Comment (cockpituous) [inline src/dialogs/permissions.jsx:105]: These 3 added lines are not executed by any test.

**Review by tomasmatus — APPROVED** [review]

**Pull request merged**